lib/services/page-service.ts

```diff
@@ -0,0 +1,152 @@
+import util from "../shared/util";
+
+// Pages backup data
+const BACKUP = require("../../consts/pages-backup-data");
+
+export class PageService implements IPageService {
+    getPages() {
+        return new Promise((resolve, reject) => {
+            resolve(BACKUP.fallback);
+        });
+    }
+
+    getFlavor(appPath: string) {
+        return new Promise((resolve, reject) => {
+            this.getAppPath(appPath)
+                .then((validPath: string) => {
+                    return this.readAppsPackageJson(validPath);
+                })
+                .then((packageJson: string) => {
+                    const ng = packageJson.indexOf("nativescript-angular") > -1;
+                    const ts = packageJson.indexOf("nativescript-dev-typescript") > -1;
+
+                    switch (true) {
+                        case ng :
+                            resolve("Angular & TypeScript");
+                            break;
+                        case ts :
+                            resolve("TypeScript");
+                            break;
+                        default:
+                            resolve("JavaScript");
+
+                    }
+                })
+                .catch((promiseError: any) => {
+                   reject(promiseError);
+                });
+        });
+    }
+
+    addPage(pageName: string, appPath: string, pageTemplate: any) {
+        return new Promise((resolve, reject) => {
+            const displayName = pageTemplate.displayName.toLowerCase();
+
+            util.pageExists(appPath, pageName)
+                .then((pageExists: boolean) => {
+                    if (!pageExists) {
+                        return this.downloadPage(displayName, pageTemplate.flavor);
+                    } else {
+                        resolve(`Page with the name: ${pageName} already exists`);
+                    }
+                })
+                .then((downloadPath: any) => {
+                // TODO Add ejs template rendering here
+                resolve("success");
+                })
+                .catch((promiseError) => {
+                    reject(promiseError);
+                });
+        });
+    }
+
+    private getAppPath(appPath: string) {
+        return new Promise((resolve, reject) => {
+            if (!util.path.isAbsolute(appPath)) {
+                reject("Path must be absolute");
+            }
+
+            appPath = util.path.normalize(appPath);
+
+            util.fs.lstat(appPath, (statErr: any, stats: any) => {
+                if (statErr) {
+                    reject(statErr);
+                }
+
+                if (!stats.isDirectory()) {
+                    reject("Not a valid app folder!");
```

Comment:
As per guidelines it is useful to make rejection reason an instanceof Error (for debugging purposes and selective error catching).

Reply:
This is a good idea. Error which are not part of the node api error response should be instance of Error